### PR TITLE
net-libs/xdp-tools: force ld.bfd, memleak fix, general ebuild improvement

### PR DESCRIPTION
**Pull request by hhoffstaette**
Three improvements:
- use sed instead of a patch to remove -Werror - easier to maintain
- force ld.bfd due to an embedding hack that would need special care with lld/mold
- add a fix for a memory leak in libxdp

This is only for 1.4.1, no need for messing with stable 1.4.0.
 
Please note that this needs a fixed toolchain-funcs.eclass which currently has a bug wrt. handling of LD: [commit](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=3dae41abc9ea4c6d642ece7ce5bc54c0548b56a5)

### CODE REVIEW

File: net-libs/xdp-tools/files/1.4.1-fix-memory-leak-in-xsk_setup_xdp_prog.patch [hunk 1]
```diff
@@ -0,0 +1,34 @@
+
```
Comment (thesamesam) [inline net-libs/xdp-tools/files/1.4.1-fix-memory-leak-in-xsk_setup_xdp_prog.patch:1]: This changes the installed program / runtime behaviour and hence needs a new revision.

File: net-libs/xdp-tools/xdp-tools-1.4.1.ebuild [hunk 2]
```diff
@@ -33,22 +33,36 @@ QA_PREBUILT="usr/lib/bpf/*.o"
 MAKEOPTS+=" V=1"
 
 PATCHES=(
-	"${FILESDIR}"/1.4.1-no-Werror.patch
+	"${FILESDIR}"/1.4.1-fix-memory-leak-in-xsk_setup_xdp_prog.patch
 )
 
+src_prepare() {
+	# remove -Werror: #899744
+	sed -i 's/-Werror//g' lib/Makefile lib/defines.mk
```
Comment (thesamesam) [inline net-libs/xdp-tools/xdp-tools-1.4.1.ebuild:41]: Missing `|| die` on external commands (all the seds here).

File: net-libs/xdp-tools/files/1.4.1-fix-memory-leak-in-xsk_setup_xdp_prog.patch (hunk 1 above)
Comment (hhoffstaette) [reply net-libs/xdp-tools/files/1.4.1-fix-memory-leak-in-xsk_setup_xdp_prog.patch:1]: :heavy_check_mark:

File: net-libs/xdp-tools/xdp-tools-1.4.1.ebuild (hunk 2 above)
Comment (hhoffstaette) [reply net-libs/xdp-tools/xdp-tools-1.4.1.ebuild]: :heavy_check_mark:

File: net-libs/xdp-tools/xdp-tools-1.4.1.ebuild [hunk 3]
```diff
@@ -33,22 +33,36 @@ QA_PREBUILT="usr/lib/bpf/*.o"
...
+
+	default
+}
+
 src_configure() {
+	# filter LTO: #861587
+	filter-lto
+
+	# filter LDFLAGS some more: #916591
+	filter-ldflags -Wl,--{icf,lto}*
```
[the remark this answers is not in the document]
Comment (hhoffstaette) [reply net-libs/xdp-tools/xdp-tools-1.4.1.ebuild:53]: :clown_face:

**Pull request closed without merging**